prow/cmd/deck/main.go

```diff
@@ -546,6 +546,11 @@ func prodOnlyMain(cfg config.Getter, pluginAgent *plugins.ConfigAgent, o options
 			logrus.Info("Manager stopped gracefully.")
 		}
 	}()
+	mgrSyncCtx, mgrSyncCtxCancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer mgrSyncCtxCancel()
```

Comment:
Please use the interrupt defer so that cleanup tasks can still use the client if they need to @alvaroaleman

Reply:
This contexts sole purpose is to give a timeout for the syncing and its being closed so we do not leak goroutines. The `defer` was just the lazy version of `call it regardless of sync success`.

If this doesn't work, the client can't be used anyways.